eng/scripts/Compile-Changelog.ps1

```diff
@@ -0,0 +1,711 @@
+
+    [Parameter(Mandatory = $false)]
+    [string]$ChangelogPath,
+
+    [Parameter(Mandatory = $false)]
+    [string]$ChangelogEntriesPath,
+
+    [Parameter(Mandatory = $false)]
+    [string]$Version,
+
+    [Parameter(Mandatory = $false)]
+    [switch]$DryRun,
+
+    [Parameter(Mandatory = $false)]
+    [switch]$DeleteFiles
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+
+# Set default paths based on ServerName
+if (-not $ChangelogPath) {
+    $ChangelogPath = "servers/$ServerName/CHANGELOG.md"
+}
+if (-not $ChangelogEntriesPath) {
+    $ChangelogEntriesPath = "servers/$ServerName/changelog-entries"
+}
+
+# Helper function to convert text to title case (capitalize first letter of each word)
+function ConvertTo-TitleCase {
+    param([string]$Text)
+    
+    if ([string]::IsNullOrWhiteSpace($Text)) {
+        return $Text
+    }
+    
+    # Use TextInfo for proper title casing
+    $textInfo = (Get-Culture).TextInfo
+    return $textInfo.ToTitleCase($Text.ToLower())
+}
+
+# Helper function to format a changelog entry with description and PR link
+function Format-ChangelogEntry {
+    param(
+        [string]$Description,
+        [int]$PR
+    )
+    
+    # Trim leading and trailing whitespace from the entire description
+    $Description = $Description.Trim()
+    
+    # Check if description contains multiple lines
+    if ($Description.Contains("`n")) {
+        # Remove trailing newlines from YAML block scalars
+        $cleanedDescription = $Description.TrimEnd("`n", "`r")
+        $lines = $cleanedDescription -split "`n"
+        $formattedLines = @()
+        
+        for ($i = 0; $i -lt $lines.Length; $i++) {
+            $line = $lines[$i].TrimEnd()  # Trim trailing spaces from each line
+            
+            if ($i -eq 0) {
+                # First line: main bullet point - also trim leading spaces
+                $line = $line.TrimStart()
+                # Ensure it ends with colon if followed by a list, otherwise period
+                if ($lines.Length -gt 1 -and $lines[1].TrimStart() -match '^-\s+') {
+                    # Next line is a bullet, so first line should end with colon
+                    if (-not $line.EndsWith(":")) {
+                        $line += ":"
+                    }
+                }
+                else {
+                    # Regular multi-line text, ensure period at end
+                    if (-not $line.EndsWith(".")) {
+                        $line += "."
+                    }
+                }
+                $formattedLines += "- $line"
+            }
+            elseif ($i -eq $lines.Length - 1) {
+                # Last line: add PR link here
+                # Line is already trimmed at the end, preserve leading indentation
+                $trimmedLine = $line.TrimStart()
+                $leadingSpaces = $line.Length - $trimmedLine.Length
+                $indent = if ($leadingSpaces -gt 0) { $line.Substring(0, $leadingSpaces) } else { "" }
```

Comment:
powershell supports string repetition through the multiply operator:

```suggestion
                $indent = ' ' * $leadingSpaces
```

Reply:
nm... considering your `$leadingSpaces`, it's safer to pull whitspace characters from $line by character count rather than using $leadingSpaces number of space characters